app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt

```diff
@@ -36,62 +49,167 @@ import com.google.firebase.Timestamp
  * @param noteViewModel The ViewModel that provides the current note to be edited and handles note
  *   updates.
  */
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun EditNoteScreen(navigationActions: NavigationActions, noteViewModel: NoteViewModel) {
   val note by noteViewModel.note.collectAsState()
-  var updatedNoteText by remember { mutableStateOf(note?.content ?: "") } // Keep track of changes
-  var updatedNoteTitle by remember { mutableStateOf(note?.title ?: "") } // Keep track of changes
+  val currentYear = Calendar.getInstance().get(Calendar.YEAR)
+  var updatedNoteText by remember { mutableStateOf(note?.content ?: "") }
+  var updatedNoteTitle by remember { mutableStateOf(note?.title ?: "") }
+  var updatedClassName by remember { mutableStateOf(note?.noteClass?.className ?: "") }
+  var updatedClassCode by remember { mutableStateOf(note?.noteClass?.classCode ?: "") }
+  var updatedClassYear by remember { mutableIntStateOf(note?.noteClass?.classYear ?: currentYear) }
+  var visibility by remember { mutableStateOf(visibilityToString(note?.public ?: false)) }
+  var expandedVisibility by remember { mutableStateOf(false) }
 
-  Column(
-      modifier = Modifier.fillMaxSize().padding(16.dp),
-      verticalArrangement = Arrangement.spacedBy(8.dp),
-      horizontalAlignment = Alignment.CenterHorizontally) {
-        Text("Edit Note Screen", modifier = Modifier.testTag("EditNote text"))
+  Scaffold(
+      modifier = Modifier.testTag("editNoteScreen"),
+      topBar = {
+        TopAppBar(
+            title = {
+              Row(
+                  modifier = Modifier.fillMaxWidth(),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    Spacer(modifier = Modifier.weight(1.5f))
+                    Text("Edit note", Modifier.testTag("editNoteTitle"))
+                    Spacer(modifier = Modifier.weight(2f))
+                  }
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() }, Modifier.testTag("goBackButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Outlined.ArrowBack,
+                        contentDescription = "Back")
+                  }
+            })
+      },
+      content = { paddingValues ->
+        LazyColumn(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(16.dp)
+                    .padding(paddingValues)
+                    .testTag("editNoteColumn"),
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              item {
+                OutlinedTextField(
+                    value = updatedNoteTitle,
+                    onValueChange = { updatedNoteTitle = it },
+                    label = { Text("Note Title") },
```

Comment:
We should raise an error if the new note title is Null. For example by creating a:
 var titleError by remember { mutableStateOf<String?>(null) }
It should be checked in the onValueChange:
 titleError = if (title.isBlank()) "Title cannot be empty" else null